### ISSUE TITLE
Warning: "Unexpected token TK_HTMLTAG found as part of a title section"

### ISSUE THREAD

**Issue by Albrecht-S** [doxygen/doxygen#9628]
**Describe the bug**
When generating the documentation of the [FLTK project](https://www.fltk.org) a bogus warning "`Unexpected token TK_HTMLTAG found as part of a title section`" appears in current Doxygen `master`. This is a regression WRT Doxygen 1.9.4 and earlier (1.9.5 not tested).

**Expected behavior**
These warnings should not appear. The token "`TK_HTMLTAG`" does not exist in the FLTK source code but I found it in the Doxygen code.

**Screenshots**
No screenshots. Warnings appearing with current FLTK documentation:
```
.../preface.dox:74: warning: Unexpected token TK_HTMLTAG found as part of a title section
.../preface.dox:74: warning: Unexpected token TK_HTMLTAG found as part of a title section
.../preface.dox:80: warning: Unexpected token TK_HTMLTAG found as part of a title section
.../preface.dox:80: warning: Unexpected token TK_HTMLTAG found as part of a title section
.../preface.dox:80: warning: Unexpected token TK_HTMLTAG found as part of a title section
.../preface.dox:80: warning: Unexpected token TK_HTMLTAG found as part of a title section
```
Link to original source file [preface.dox](https://github.com/fltk/fltk/blob/4d17cf310dc953703bf3a60763b681eb217bc3e9/documentation/src/preface.dox#L74). Note that the preceding lines of those mentioned in the warnings contain HTML `<tt>` statements. I'll follow-up with a partial workaround (removing these `<tt>` tags).

**To Reproduce**
I'm currently not able to create a small reproducer since the warnings are not related to our FLTK source code. However, it's easy to reproduce the warnings with current FLTK code:
```
git clone https://github.com/fltk/fltk.git
cd fltk
# optionally check out `master` as of today:
git checkout 4d17cf310dc953703bf3a60763b681eb217bc3e9
mkdir build
cd build
# Make sure that `doxygen` can be found in your path so cmake finds it.
# I assume you know how to do this, but anyway:
# example: export PATH=/path/to/doxygen/build/bin:$PATH
cmake ..
make html
```
Please ignore warnings like "Tag 'CLASS_DIAGRAMS' ... has become obsolete. These warnings are due to an older Doxyfile (as you certainly know).

**Version**
$ doxygen --version
1.9.6 (59aa82c1385054dadfb636375e7e45be6b95672c)
(self built)

**Additional context**
I'll try to `git bisect` when time permits and follow-up with more information.

**Comment by Albrecht-S** [comment]
Further testing revealed that these warnings can be reduced but not entirely eliminated by removing the HTML `<tt>` tags from the mentioned paragraphs, resulting in the following code with line numbers:
```
70  \par MS Windows, WIN32
71  The Microsoft Windows Application Programmer's Interface for Windows 2000,
72  Windows XP, Windows Vista, Windows 7 and later Windows versions.
73  FLTK uses the preprocessor definition _WIN32 for the 32 bit
74  and 64 bit MS Windows API.
75  
76  \par OS X, __APPLE__
77  The Apple desktop operating sytem OS X 10.0 and later. MacOS 8 and 9 support
78  was dropped after FLTK 1.0.10. FLTK uses the preprocessor definition
79  __APPLE__ for OS X.
80  
```
This results in only two warnings:
```
.../preface.dox:80: warning: Unexpected token TK_HTMLTAG found as part of a title section
.../preface.dox:80: warning: Unexpected token TK_HTMLTAG found as part of a title section
```

Removing **all** html tags from this file does not eliminate the remaining two warnings.

Tested with Doxygen `1.9.6 (59aa82c1385054dadfb636375e7e45be6b95672c)`.

**Comment by albert-github** [comment]
The fact that in case of `__APPLE__` we still get the warning is because doxygen will translate this to `<strong>APPLE</strong>`, not sure whether or not in your documentation the `<strong>` of the word `APPLE` is intended or the word `__APPLE__`. In the later case you can use here: `\__APPLE__`.

The problem with the `<tt>` ca easily be reproduced with just the example:
```
 \page preface Preface

\par MS Windows, <tt>WIN32</tt>
```
strangely enough the removal of the `\page` line will also remove the warning.

(used `Doxyfile`, just an empty file)

**Comment by Albrecht-S** [comment]
@albert-github Thanks for checking this. In case of `__APPLE__` we **mean** the preprocessor constant defined by Apple (macOS) systems. However, I see that it isn't displayed as intended. Thanks for the '\__APPLE__' escape hint, I'll try this where appropriate.

But anyway, this issue (regarding `<tt>` and other HTML tags) is a regression introduced between 1.9.4 and 1.9.5. I'm just bisecting and will follow-up with more info.

**Comment by Albrecht-S** [comment]
Git bisect issued:
```
$ git bisect bad
b3bd30ff95b9630cdbc5b592e82b95c314531eeb is the first bad commit
commit b3bd30ff95b9630cdbc5b592e82b95c314531eeb
Author: albert-github <albert.tests@gmail.com>
Date:   Tue Jul 26 18:01:02 2022 +0200

    bug 786576 regression with ALIASES or image latex interpretation
    
    The problem is not the ALIAS but ... (more text elided)
```

This particular commit issues even more warnings which must have been fixed meanwhile:
```
/git/fltk/master/documentation/src/preface.dox:74: warning: Unexpected token TK_HTMLTAG found as part of a title section
/git/fltk/master/documentation/src/preface.dox:74: warning: Unexpected token TK_HTMLTAG found as part of a title section
/git/fltk/master/documentation/src/preface.dox:80: warning: Unexpected token TK_HTMLTAG found as part of a title section
/git/fltk/master/documentation/src/preface.dox:80: warning: Unexpected token TK_HTMLTAG found as part of a title section
/git/fltk/master/documentation/src/preface.dox:80: warning: Unexpected token TK_HTMLTAG found as part of a title section
/git/fltk/master/documentation/src/preface.dox:80: warning: Unexpected token TK_HTMLTAG found as part of a title section
/git/fltk/master/FL/Fl_Counter.H:37: warning: Unexpected token TK_HTMLTAG found as part of a image
/git/fltk/master/FL/Fl_File_Input.H:37: warning: Unexpected token TK_HTMLTAG found as part of a image
/git/fltk/master/src/Fl_Help_Dialog_Dox.cxx:26: warning: Unexpected token TK_HTMLTAG found as part of a image
/git/fltk/master/FL/Fl_Light_Button.H:33: warning: Unexpected token TK_HTMLTAG found as part of a image
/git/fltk/master/FL/Fl_Menu_Button.H:29: warning: Unexpected token TK_HTMLTAG found as part of a image
/git/fltk/master/FL/Fl_Output.H:32: warning: Unexpected token TK_HTMLTAG found as part of a image
/git/fltk/master/FL/Fl_Positioner.H:32: warning: Unexpected token TK_HTMLTAG found as part of a image
/git/fltk/master/FL/Fl_Round_Button.H:26: warning: Unexpected token TK_HTMLTAG found as part of a image
/git/fltk/master/FL/Fl_Simple_Counter.H:26: warning: Unexpected token TK_HTMLTAG found as part of a image
/git/fltk/master/FL/Fl_Valuator.H:38: warning: Unexpected token TK_HTMLTAG found as part of a image
```

I hope this helps finding the bug.

I can try to fix the `__APPLE__` case in our source code.

However, regarding the usage of `<tt>` and other HTML tags "as part of a title section": is there a reason why this should be disallowed or is this warning a Doxygen bug (regression)?

**Comment by albert-github** [comment]
You are bringing in a new warning in your last comment:
```
warning: Unexpected token TK_HTMLTAG found as part of a image
```
is this last warning also visible with the current master?
(with some 1.9.5 intermediate executable I can see that warning but not with the 1.9.5 version nor with the current master in my test program. anyway it is also strange to position the html image in a center paragraph more appropriate would be a `<div>` but I don't think it is necessary at all, an image is packed inside a `<div class="image">` so it would be better to handle it in the CSS file and the default is "center. To obtain an inline image one can use the option `inline` like in `\image{inline}`)

When you still see the image problem or a similar problem:
- please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent. The `fltk` repository is a moving target.

**Comment by albert-github** [comment]
The question:
> However, regarding the usage of `<tt>` and other HTML tags "as part of a title section": is there a reason why this should be disallowed or is this warning a Doxygen bug (regression)?

At first glance there is no reason not allowing it, but it has to be investigated especially regarding the non style change JTML commands (stye change commands are a.o.a `<tt>`, `<b>`)
When we look at the results of 1.9.4 the `<tt>` etc. are just not present in the output, so in this respect the warning is unexpected but OK (setting it to enhancement).

**Comment by Albrecht-S** [comment]
@albert-github I wrote:
> This particular commit issues even more warnings which must have been fixed meanwhile

which might not be absolutely clear, sorry for the confusion. I meant that these additional warnings "`TK_HTMLTAG found as part of a image`" appear(ed) with this specific commit but they do no longer exist in master. I've seen them in more than one commit (of about 8 or 9 while bisecting) but I didn't check when they disappeared.

> it is also strange to position the html image in a center paragraph ...

I didn't check these particular warnings related to images because they don't exist in later Doxygen versions. But I appreciate all your hints and I'll look if there's anything wrong or can be done better. Our Doxygen documentation was created in a large team effort around 2008 and later, hence there can have been inconsistent "strategies" of embedding images.

> Please don't add external links as they might not be persistent. The fltk repository is a moving target.

Thanks, I'm aware of this problem. The links I posted are "permanent links" to specific versions of the code and for the FLTK clone/checkout I added a specific commit for exactly that reason.

> When you still see the image problem or a similar problem ...

The only issues (bogus warnings) I see now in current master (as tested by me, i.e. the commit mentioned in my initial post) are those mentioned in the that post. You found the minimal reproducer yourself (thanks!) and this should be sufficient for further testing. TIA.

If I can do anything else please let me know.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9630

**Comment by Albrecht-S** [comment]
@albert-github Thank you for the quick fix.

I confirm that b553af3975c2d66489a11c69d923f3ba4864a06d fixes the warnings. I built the documentation of several FLTK versions w/o issues (no such warnings).

Looking forward to the merge...

**Comment by Albrecht-S** [comment]
FTR: I updated to 70d0e804d9f7d0bac2685759de1ed42e2fdbf2a3 and confirm that this also works fine for me (FLTK docs).

**Comment by Albrecht-S** [comment]
The "TK_HTMLTAG" warnings have apparently been fixed in 1.9.6 (git Release-1.9.6) although I don't see that the related PR #9630 has been merged. I'm closing this issue now because my issue has been fixed. I'm leaving #9630 to the maintainers.

Thanks for fixing this issue.

**Comment by andy-nelson-mw** [comment]
We've experienced the "TK_HTMLTAG" warning in Doxygen 1.9.5 simply by having an extra space at the end of a "@par" line, ie: /* @par My Title  */  (github's comment viewer of course reduces those 2 spaces after "Title" down to 1.)

**Comment by albert-github** [comment]
@andy-nelson-mw 

Doxygen 1.9.5 is older the current doxygen version is 1.15.0 and as noted in https://github.com/doxygen/doxygen/issues/9628#issuecomment-1366076214it should heave been fixed in 1.9.6.
If it still appears with the current doxygen version, please open a new issue and attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem?
   Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).
